config/logging_level.go

```diff
@@ -0,0 +1,70 @@
+package config
+
+import (
+	"sync"
+
+	"github.com/edaniels/golog"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var globalLogger struct {
+	// These variables are initialized once at startup. No need for special synchronization.
+	logger           golog.Logger
+	cmdLineDebugFlag bool
+	logLevel         zap.AtomicLevel
+
+	// These variables can be changed while the `viam-server` is running. Additionally, every time one
+	// of these is changed, we re-evaluate the log level. This mutex synchronizes the reads and writes
+	// that can concurrently happen.
+	mu                   sync.Mutex
+	fileConfigDebugFlag  bool
+	cloudConfigDebugFlag bool
+	currentLevel         zapcore.Level
+}
+
+func InitLoggingSettings(logger golog.Logger, cmdLineDebugFlag bool, logLevel zap.AtomicLevel) {
+	globalLogger.logger = logger
+	globalLogger.cmdLineDebugFlag = cmdLineDebugFlag
+	globalLogger.logLevel = logLevel
+	globalLogger.currentLevel = logLevel.Level()
+	globalLogger.logger.Info("Log level initialized: ", globalLogger.currentLevel)
+}
+
+func UpdateFileConfigDebug(fileDebug bool) {
```

Comment:
I don't think `UpdateFileConfigDebug` and `UpdateCloudConfigDebug` could ever feasibly run in parallel (I'm not aware of an ability for the RDK to run both off a cloud and local config). Nor do I think that multiple calls of either function could happen simultaneously. The locking still seems like a good safety measure, though.